tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/OnlineShopDFDCyclicTest.java

```diff
@@ -0,0 +1,151 @@
+package org.dataflowanalysis.analysis.tests.dfd;
+
+import static org.dataflowanalysis.analysis.tests.AnalysisUtils.TEST_MODEL_PROJECT_NAME;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertIterableEquals;
+
+import java.nio.file.Paths;
+import java.util.List;
+import org.dataflowanalysis.analysis.core.AbstractVertex;
+import org.dataflowanalysis.analysis.core.DataCharacteristic;
+import org.dataflowanalysis.analysis.dfd.DFDConfidentialityAnalysis;
+import org.dataflowanalysis.analysis.dfd.DFDDataFlowAnalysisBuilder;
+import org.dataflowanalysis.analysis.dfd.core.DFDCharacteristicValue;
+import org.dataflowanalysis.analysis.dfd.core.DFDFlowGraphCollection;
+import org.dataflowanalysis.analysis.dfd.core.DFDVertex;
+import org.dataflowanalysis.examplemodels.Activator;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class OnlineShopDFDCyclicTest {
```

Comment:
Is this class a full copy of `OnlineShopDFDTest` with one added method `testIsNotCyclic()`? Couldn't this just be added to the original test? We should minimize code duplication, also in test classes.

Reply:
yes, but eventually they run on different analysis, since they should provide different TFGFinders. But did not figure out, how to properly work with a ressourceprovider

Reply:
Ah, I understand the problem but I'm still not happy with the code duplication 😁 Let's discuss how to solve this tommorow